torch/distributed/_tensor/api.py

```diff
@@ -422,14 +422,16 @@ def redistribute(
         if placements is None:
             raise RuntimeError("placements is needed for redistribute!")
 
-        for placement in placements:
+        placements = list(placements)
```

Comment:
Another suggestion here is that maybe we can create a new list here instead of casting a potential tuple to list or modify the user passed in list, wdyt?

Reply:
If I understand correctly, calling `list()` on an existing list creates a shallow copy. Shallow copy seems like the desired behavior since we only _replace_ negative-dimension `Shard` placements but not modify them in place, so we can reuse as much of the user's `Placement` objects as possible? Let me know what you think.

```
>>> class A:  # define some arbitrary class
>>>   def __init__(self):
>>>     self.a = 1
>>> l = [A(), A()]
>>> id(l)
140259468644864
>>> id(l[0])
140259469146912
>>> id(l[1])
140259469415216
>>> ll = list(l)
>>> id(ll)
140259468666560  <-- different
>>> id(ll[0])
140259469146912  <-- same
>>> id(ll[1])
140259469415216  <-- same
```

Reply:
Got it, yeah then we don't have concern that we modified the user passed in list, that convinced me!